### PR TITLE
A number of tweaks to make the button-press handling more consistent (CU-f0w9qm)

### PR DESCRIPTION
**Pull request by trisapeace**
It might be easiest to go through this PR by going through the commits rather than looking at all the code at once.

There are four main changes:

1. When there is a **double click**, Power Automate / Flow sends the parameter `Type=double click`. However, in the code, we were looking for `double_click` (with an underscore). This means that when we received a request with `Type=double click`, we were treating it the same as a single click. This means that we might have missed the initial message for some urgent requests (they would still have received the reminder and fallback messages).

2. When a request comes in to `POST /` from Power Automate / Flow, this now counts it as only 0.5 of a button press. This will hopefully band-aid over the problem where it always **sends us two copies of each request**. This required a change in the DB to allow decimal values in the `num_presses` column, but I think it works. Unfortunately, at this time there are 38 rows in `sessions` that have an odd value in the `num_presses` column that are not the Princess Rooms (10 of them have `num_presses = 1`). This is out of the 1794 non-Princess Rooms sessions in that table.

3. **No longer require the `button-battery-level` header** in requests to `POST /flic_button_press`. I've seen cases where the Flic Internet Request doesn't send the button-battery-level header but everything else looks OK. I don't want to miss these presses, since there is no way to inform the resident that the button press did not result in a message. Therefore, this now allows these requests and just not updating the DB's button_battery_level value.

4. Now that we are counting each `POST /` as half a press. We can trust that the `num_presses` value is correct and should sent an **urgent text message when `numPresses == 2`**.

There are some other little changes to move things around, improve transaction handling, add/update tests, removing nearby semi-colons, etc.

This is available on `chatbot-dev` for testing.

### CODE REVIEW

**Comment by trisapeace** [comment]
Task linked: [CU-f0w9qm Improve logic of when to send initial and urgent text messages](https://app.clickup.com/t/f0w9qm)

File: server/server.js
Comment (trisapeace) [inline server/server.js:317]: @schwarrrtz @mariocimet 

This change is the only really important one. I think that we should get this deployed sooner rather than later because without it, we end up the situation where if the resident's first press is considered "numPresses = 2", then we don't send an initial message. It's only an issues at the Princess Rooms.

**Comment by schwarrrtz** [comment]
re. "This means that when we received a request with `Type=double click`, we were treating it the same as a single click. This means that we might have missed the initial message for some urgent requests (they would still have received the reminder and fallback messages)."

good catch - but wouldn't they just have been downgraded to non-urgent presses? not lost entirely

File: server/server.js
Comment (darvell) [inline server/server.js:349]: If things can sometimes be a bit iffy, why not req.body.Type.startsWith('double') so it's caught no matter what?

**Comment by trisapeace** [comment]
@schwarrrtz You're right. I think I was getting it conflated with the problem of when `presses=2` comes in as the first request to `/flic_button_press`. Changed description to say the real consequences ("This means that we would not get an urgent message when we should have.")

File: server/server.js
Comment (trisapeace) [reply server/server.js]: Thanks @darvell , good idea. 

Done!

**Review by mariocimet — APPROVED** [review]

File: server/server.js
Comment (trisapeace) [inline server/server.js:346]: @trisapeace Change this back.

**Review by schwarrrtz — APPROVED** [review]
After reviewing this myself & also in a meeting with Theresa and Mario I think we should approve this. I'd still like to test it myself on `chatbot-dev` before we deploy it. We also noted during the meeting that once we merge & deploy this we should get the old installations onto the Flic Internet Request system relatively soon.

File: server/server.js
Comment (trisapeace) [reply server/server.js]: Done!

**Comment by schwarrrtz** [comment]
OK i've just finished up testing this on `chatbot-dev` and it seems to work great. Nice work @trisapeace.

**Pull request merged**
